timesketch/api/v1/resources/archive.py

```diff
@@ -481,39 +628,153 @@ def _archive_sketch(self, sketch: Sketch):
         for label in labels_to_prevent_deletion:
             if sketch.has_label(label):
                 abort(
-                    HTTP_STATUS_CODE_FORBIDDEN,
-                    "A sketch with the label {:s} cannot be " "archived.".format(label),
+                    HTTP_STATUS_CODE_BAD_REQUEST,
+                    f"Sketch {sketch.id} has label '{label}'"
+                    " and cannot be archived.",
                 )
 
-        sketch.set_status(status="archived")
+        # Check if any timeline is in a non-archivable state
+        non_archivable_timeline_statuses = ["processing", "fail", "timeout"]
+        for timeline_to_check in sketch.timelines:
+            timeline_status = timeline_to_check.get_status.status
+            if timeline_status in non_archivable_timeline_statuses:
+                base_error_msg = (
+                    f"Cannot archive sketch {sketch.id}. Timeline "
+                    f"'{timeline_to_check.name}' (ID: {timeline_to_check.id}) is in "
+                    f"a non-archivable state: '{timeline_status}'."
+                )
+                suggestion = ""
+                if timeline_status in ["fail", "timeout"]:
+                    suggestion = " Please delete this timeline and try again."
+                elif timeline_status == "processing":
+                    suggestion = (
+                        " Please wait for the timeline to finish processing or "
+                        "delete it."
```

Comment:
iirc, you can't delete a processing timeline. So only way is to wait for it to finish. Which makes sense, since (as long as its not a bug) the search index is importing stuff during processing status.

Reply:
there might be instances where it is indef in preocessing state, that is why the recomendation to delete.

Reply:
hm I think I can add the message we write to the logs to be adde to the user facing issue here